benchmark/utils.py

```diff
@@ -0,0 +1,33 @@
+"""
+Helper methods for benchmarking
+
+Created on Fri Jun 19 03:14:44 2020
+
+@author: Henry Liu
+"""
+
+import numpy as np
+import tensorflow as tf
+
+with tf.device("GPU:0"):
+    g1 = tf.random.Generator.from_non_deterministic_state()
+    g2 = g1.split()[0]
+
+
+def random_alphas(num):
+    phases = tf.complex(
+        tf.constant(0, dtype=tf.float32),
+        g1.uniform(
+            [num], maxval=tf.constant(np.pi, dtype=tf.float32), dtype=tf.float32
+        ),
+    )
+    amplitudes = tf.cast(
+        g2.uniform([num], maxval=tf.constant(5, dtype=tf.float32), dtype=tf.float32),
+        dtype=tf.complex64,
+    )
+    return amplitudes * tf.math.exp(phases)
+
+
+def creation(N):
```

Comment:
In qutip this is called qt.destroy, I think we should rename this to "destroy" to avoid confusion (because technically this is a destruction operator, not creation operator. Creation operator is adjoint of this thing).  Also, the way you generate it works fine in tf2.2 (hopefully, I didn't check it explicitly) but it has a bug in tf2.1 -- the argument k=1 which specifies the offset of the diagonal doesn't work, so in tf2.1 this generates a tensor with main diagonal. Can you add a check for the tensorflow version here and generate it in a different way for tf2.1. I still haven't switched to tf2.2

Reply:
Ah, right - I saw your comment about the k=1 bug. Sounds good

Reply:
https://github.com/tensorflow/tensorflow/issues/35761

Reply:
https://github.com/tensorflow/tensorflow/issues/35761